src/cart/crt_rpc.c

```diff
@@ -1304,6 +1304,30 @@ crt_req_abort(crt_rpc_t *req)
 	return rc;
 }
 
+uint64_t
+crt_hlc_send_get(crt_rpc_t *req)
+{
+	struct crt_rpc_priv	*rpc_priv;
+
+	D_ASSERT(req != NULL);
+
+	rpc_priv = container_of(req, struct crt_rpc_priv, crp_pub);
+
+	return rpc_priv->crp_req_hdr.cch_hlc;
+}
+
+uint64_t
+crt_hlc_receive_get(crt_rpc_t *req)
```

Comment:
probably can check crp_state, for example if the RPC is still in flight then the returned hlc is non-sense. (return error or just and a log or err msg?)
or in API comment explicitly say returned hlc is un-trustable if the RPC is still inflight.

Reply:
My assumption is you will use this API in completion callback of RPC. So, you will have an information about RPC status. I'm not in favor of block this call until RPC completion or check for status.

I also assume there are no errors can be in this call but it can return zero instead of correct HLC timestamp. From my point of view this is fine because of HLC should be used to provide correct order. Zero timestamp is always less all other timestamps.